src/sql/workbench/common/editorReplacerContribution.ts

```diff
@@ -55,6 +56,11 @@ export class EditorReplacementContribution implements IWorkbenchContribution {
 			language = this.modeService.getModeIdByFilepathOrFirstLine(editor.getResource());
 		}
 
+		if (!language) {
+			// just use the extension
+			language = path.extname(editor.getResource().toString()).slice(1); // remove the .
```

Comment:
how does this work if the file doesn't have an extension?

Reply:
I'll double check error cases but extname should be and empty string so slice will return and empty string.

Reply:
I've verified that the extname method returns an empty string if the string has no extname, then slice(1) will just return an empty string again.